app/src/main/java/com/android/unio/ui/event/EventCreation.kt

```diff
@@ -188,7 +220,48 @@ fun EventCreationScreen(
 
           Button(
               modifier = Modifier.testTag(EventCreationTestTags.SAVE_BUTTON),
-              onClick = { navigationAction.goBack() }) {
+              enabled =
+                  name.isNotEmpty() &&
+                      shortDescription.isNotEmpty() &&
+                      longDescription.isNotEmpty() &&
+                      startTimestamp != null &&
+                      endTimestamp != null &&
+                      startTimestamp!! < endTimestamp!! &&
+                      eventBannerUri.value != Uri.EMPTY,
+              onClick = {
+                val inputStream = context.contentResolver.openInputStream(eventBannerUri.value)!!
+                eventViewModel.addEvent(
+                    inputStream,
+                    Event(
+                        uid = "", // This gets overwritten by eventViewModel.addEvent
+                        title = name,
+                        organisers =
+                            Association.firestoreReferenceListWith(
+                                (coauthorsAndBoolean
+                                        .filter { it.second.value }
+                                        .map { it.first.uid } +
+                                        associationViewModel.selectedAssociation.value!!.uid)
+                                    .distinct()),
+                        taggedAssociations =
+                            Association.firestoreReferenceListWith(
+                                taggedAndBoolean.filter { it.second.value }.map { it.first.uid }),
+                        image = eventBannerUri.value.toString(),
+                        description = longDescription,
+                        catchyDescription = shortDescription,
+                        price = 0.0,
+                        startDate = startTimestamp!!,
+                        endDate = endTimestamp!!,
+                        location = Location(),
```

Comment:
Just leaving this comment for future reference: This won't be an empty location when the Search Places API will be implemented (or Nominatim)